File: so/nif_demo/niftest.erl
-module(niftest).

-export([init/0, hello/0]).

init() ->

      erlang:load_nif("./niftest", 0).

hello() ->

      "NIF library not loaded".


% 1> c(niftest).

% {ok,niftest}

% 2> niftest:hello().

% "NIF library not loaded"

% 3> niftest:init().

% ok

% 4> niftest:hello().

% "Hello world!"

File: so/nif_demo/niftest.c
#include "erl_nif.h"

static ERL_NIF_TERM hello(ErlNifEnv* env, int argc, const ERL_NIF_TERM argv[])
{
	return enif_make_string(env, "Hello world!", ERL_NIF_LATIN1);
}

static ErlNifFunc nif_funcs[] =
{
	{"hello", 0, hello}
};

ERL_NIF_INIT(niftest, nif_funcs,NULL,NULL,NULL,NULL)

// https://blog.csdn.net/wqtn22/article/details/84254874
// gcc -fPIC -shared -o niftest.so niftest.c -I $ERL_ROOT/usr/include/
// /usr/local/erlang_18.3/lib/erlang/usr/include
// gcc -fPIC -shared -o niftest.so niftest.c -I /usr/local/erlang_18.3/lib/erlang/usr/include